src/components/CouponsChart/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState } from "react";
+import "./style.css";
+import { Bar } from "react-chartjs-2";
+import { Typography, Divider, Statistic, Row, Col, Card, Spin } from "antd";
+import { useTranslation } from "react-i18next";
+import firebase from "../../firebaseConfig";
+
+const db = firebase.firestore();
+const { Title } = Typography;
+
+const CouponsChart = () => {
+  const { t } = useTranslation();
+  const [soldCoupons, setSoldCoupons] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const unsubscribe = db
+      .collection("soldCoupons")
+      .where("compName", "==", "Arby's")
+      .onSnapshot((snapshot) => {
+        const dataArr = [];
+        snapshot.forEach((doc) => {
+          dataArr.push({ ...doc.data(), docId: doc.id });
+        });
+        setSoldCoupons(dataArr[0]);
```

Comment:
u are getting all the sold coupons and then only using the first one to set the state why is that ?

Reply:
No, I'm not getting all of them I have a where clause and only getting one of them. So, there is only one doc that I'm using which is inside dataArr